cmd/wallet-web/src/App.vue

```diff
@@ -5,27 +5,77 @@
 -->
 
 <script setup>
-import { onBeforeMount, onMounted, onUnmounted, ref, computed } from 'vue';
+import { onBeforeMount, onUnmounted, computed, watch } from 'vue';
+import { useRoute } from 'vue-router';
 import { useStore } from 'vuex';
 import getStartingLocale from '@/mixins/i18n/getStartingLocale.js';
 import { updateI18nLocale } from '@/plugins/i18n';
+import useBreakpoints from '@/plugins/breakpoints.js';
 import SpinnerIcon from '@/components/icons/SpinnerIcon.vue';
+import { CHAPIHandler } from '@/mixins';
 
 // Local Variables
 const startingLocale = getStartingLocale(); // Get starting locale, set it in i18n and in the store
 
 // Hooks
+const route = useRoute();
 const store = useStore();
+const breakpoints = useBreakpoints();
 
 // Store Getters
 const userLoaded = computed(() => store.getters['isUserLoaded']);
+const username = computed(() => store.getters['getGnapSubjectId']);
+const CHAPIEnabled = computed(() => store.getters['getEnableCHAPI']);
+const credentialMediatorURL = computed(() => store.getters['getCredentialMediatorURL']);
+console.log('CHAPIEnabled', CHAPIEnabled.value);
+
+// Store Actions
+const activateCHAPI = () => store.dispatch('activateCHAPI');
+
+// if intended target doesn't require CHAPI or if it is disabled
+const enableCHAPI = computed(
+  () =>
+    CHAPIEnabled.value &&
+    !route.params?.disableCHAPI &&
+    (route.name === 'chapi-worker' || (!breakpoints.xs && !breakpoints.sm))
+);
+console.log('enableCHAPI', enableCHAPI.value);
 
 store.dispatch('setLocale', startingLocale);
 
 onBeforeMount(async () => {
   await updateI18nLocale(startingLocale.id);
 });
 
+watch([enableCHAPI, () => username], async ([enableCHAPI, username]) => {
+  console.log('!breakpoints.xs', !breakpoints.xs);
+  console.log('!breakpoints.sm', !breakpoints.sm);
+  console.log('route.params', route.params);
+  console.log('!route.params?.disableCHAPI', !route.params?.disableCHAPI);
+  console.log('watch CHAPIEnabled', CHAPIEnabled.value);
+  console.log('watch enableCHAPI', enableCHAPI);
+  console.log('watch username', username);
+  if (enableCHAPI && username.value) {
+    const polyfill = await import('credential-handler-polyfill');
+    console.log('polyfill', polyfill);
+
+    const webCredentialHandler = await import('web-credential-handler');
+    console.log('webCredentialHandler', webCredentialHandler);
+
+    console.log('creating CHAPIHandler');
+    const chapi = new CHAPIHandler(polyfill, webCredentialHandler, credentialMediatorURL.value);
+    console.log('created chapi', chapi);
+
+    console.log('username', username.value);
+    chapi
+      .install(username.value)
+      .then(() => activateCHAPI())
+      .catch((e) => {
+        console.error(e);
+      });
+  }
+});
+
```

Comment:
not sure why we need CHAPI install in App.vue when we can do it after login (or init after new page/tab) triggered by vuex state.